prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchemaReader.kt

```diff
@@ -38,16 +63,31 @@ object ConfigSchemaReader : Logging {
             }
             val depSchemaFolder = "$folder/${FilenameUtils.getPath(schemaList.last().extends)}"
             val depSchemaName = FilenameUtils.getName(schemaList.last().extends)
-            schemaList.add(readSchemaTreeFromFile(depSchemaName, depSchemaFolder, schemaClass = schemaClass))
+            schemaList.add(readSchemaTreeRelative(depSchemaName, depSchemaFolder, schemaClass = schemaClass))
         }
+        return schemaList
+    }
 
-        // Now merge the parent with all the child schemas
-        val mergedSchema = mergeSchemas(schemaList)
-
-        if (!mergedSchema.isValid()) {
-            throw SchemaException("Invalid schema $schemaName: \n${mergedSchema.errors.joinToString("\n")}")
+    /**
+     * Read a schema from a [URI] using the scheme to determine who to read the schema
+     *
+     */
+    private fun fromUri(
+        schemaUri: URI,
+        schemaClass: Class<out ConfigSchema<out ConfigSchemaElement>>
+    ): List<ConfigSchema<*>> {
+        val schemaList = mutableListOf<ConfigSchema<*>>()
+        schemaList.add(readSchemaTreeUri(schemaUri, schemaClass = schemaClass))
+        while (!schemaList.last().extends.isNullOrBlank()) {
```

Comment:
Is any of this circular dependency checking needed if `readSchemaTreeUri` also does it? Can we consolidate it to one place?

Reply:
Yes I think can likely clean that up

Reply:
Ah ok, so this actually touches on what we were chatting about last week.  This is the special handling to make sure that the extending clauses don't create a circular dependency.  I think you're likely right that we could clean this up, but it might be out of scope here?